docs/features/kubernetes/troubleshooting.md

```diff
@@ -0,0 +1,98 @@
+---
+id: troubleshooting
+title: Troubleshooting Kubernetes
+sidebar_label: Troubleshooting
+description: Troubleshooting for Kubernetes
+---
+
+## Kubernetes is not showing up on Service Entities
+
+Sometimes, Kubernetes is not showing up on service entities, we should test your
+k8s cluster are already connected to Backstage or not.
+
+```curl
+curl --location --request POST '{{backstage-backend-url}}:{{backstage-backend-port}}/api/kubernetes/services/:service-entity-name' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "entity": {
+        "metadata": {
+            "name": <service-entity-name>
+        }
+    }
+}
+'
+```
+
+The curl response should have resources from Kubernetes:
+
+```json
+# curl response
+{
+  "items": [
+    {
+      "cluster": {
+        "name": <cluster-name>
+      },
+      "resources": [
+        {
+          "type": "services",
+          "resources": [
+            {
+              "metadata": {
+                "creationTimestamp": "2022-03-13T13:52:46.000Z",
+                "labels": {
+                  "app": <service-entity-name>,
+                  "backstage": <selector>,
+                  "backstage.io/kubernetes-id": <service-entity-name>
+                },
+                "name": <service-entity-name>,
+                "namespace": <namespace>
+              },
+              ....
+            }
+          ]
+        },
+        ....
+        {
+          "type": "pods",
+          "resources": [
+            ,,,,
+          ]
+        }
+      ],
+      "errors": []
+    }
+  ]
+}
+
+```
+
+Kubernetes will not be showing anything when catalog info annotations is not
+match with k8s related yaml label (service.yaml, deployment.yaml, etc). We
+recommend you for adding two labels and using label selector annotations:
+
+`backstage: <selector>` and `backstage.io/kubernetes-id: <entity-service-name>`.
```

Comment:
This seems quite strange tbh, do you really need to keep the `backstage.io/kubernetes-id` annotation in the k8s object when using a label selector?

Reply:
this one is exactly what has tripped me up; but haven't ever had time to really test it

Reply:
yup, backstage selector for annotations on catalog-info,yaml, and backstage.io/kubernetes-id for query to k8s
I print the plugin there and using labelSelector for it if params not described, the frontend in my local hit without params.

https://github.com/backstage/backstage/blob/master/plugins/kubernetes-backend/src/service/KubernetesFetcher.ts#L119

Then, the logger print this
<img width="925" alt="Screen Shot 2021-03-04 at 08 53 54" src="https://user-images.githubusercontent.com/9495683/109899158-012d9500-7cc8-11eb-8842-ca9ed83aba60.png">

But i didn't try yet when there are no annotations on catalog-info.yaml. I think its possible just using backstage.io/kubernetes-id on k8s related yaml, and not define anything on catalog-info.yaml, but it will be scan all Namespace so there are no separations between namespace, i think it will be useful for namespace separation (staging|production) 

https://github.com/backstage/backstage/blob/master/plugins/kubernetes-backend/src/service/KubernetesFetcher.ts#L206

Reply:
Namespace separation would be great, I think we can add an issue for it.

Reply:
Great! Then, should we use 2 labels or just 1 label without annotations on catalog-info.yaml for this docs ? What would you suggest ? @mclarke47 @adamdmharvey @Rugvip

Reply:
I think just 1 label is fine